#31-게시물 미리보기 구현

wjddnwls918 opened the pull request:
### The kind of change this PR does introduce

* [ ] a bug fix
* [x] a new feature
* [ ] an update to the documentation
* [ ] a code change that improves performance
* [ ] other

### Current behaviour


### New behaviour

- 게시물 미리보기를 구현했습니다.
- layout 추가, BoardPreviewDialog 클래스 추가

### Other information (e.g. related issues)

resolved #31

wjddnwls918 committed:
#31-게시물 미리보기 구현

wjddnwls918 committed:
Merge branch 'develop' into feature/#31-Board_Preview

ro0opf commented on app/build.gradle:
```diff
@@ -69,5 +69,7 @@ dependencies {
     implementation files('libs/libDaumMapAndroid.jar')
 
+    //CircleImageView
+    implementation 'de.hdodenhof:circleimageview:3.0.0'
```
버전을 gradle project 수준에서 변수로 정의해주세요.

wjddnwls918 replied:
7bb31b49f8c9cdcf295490f9a22edef7801d2643

ro0opf commented on app/src/main/java/com/teamdonut/eatto/ui/board/BoardPreviewDialog.java:
```diff
@@ -0,0 +1,50 @@
+        View v = inflater.inflate(R.layout.board_preview_dialog, container, false);
+        // remove dialog background
+        getDialog().getWindow().setBackgroundDrawable(
+                new ColorDrawable(android.graphics.Color.TRANSPARENT));
```
`Dialog dialog = getDialog();
        if(dialog != null) {
            int height = ViewGroup.LayoutParams.MATCH_PARENT;
            dialog.getWindow().setLayout(900 , height);
        }`

` setStyle(DialogFragment.STYLE_NO_TITLE, R.style.DialogTheme);`

위 같은 경우 다른 메소드에서 override하는 이유가 있나용??

wjddnwls918 replied:
위치를 여러번 변경해봤는데 
아마 Fragment의 생명주기와 연관이 있는것 같네요
onCreate에서는 UI관련 작업을 할 수 없다네요

ro0opf commented on app/src/main/java/com/teamdonut/eatto/ui/map/MapFragment.java:
```diff
@@ -54,6 +50,18 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
         binding.flMapView.addView(mapView);
 
+
+        binding.ibShowBottomSheet.setOnClickListener(new View.OnClickListener() {
```
테스트로 용도로 만들어 놓으신 버튼 리스너는 지워도 될 것 같습니다.

wjddnwls918 replied:
9ee8fe4873b928839db568137867dae9e33cf50d

ro0opf commented on app/src/main/res/layout/board_preview_dialog.xml:
```diff
@@ -0,0 +1,167 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout
```
constraintlayout 속성을 이용하여 1depth로 구성하는건 어떨까요?
(1) 상,하,좌,우 가이드라인 속성
(2) 상단 가이드 라인 아래 CircleImageView 위치
(3) CircleImageView 가운데부터 view들 위치?
@Jiyunn

ro0opf commented on app/src/main/res/layout/board_preview_dialog.xml:
```diff
@@ -0,0 +1,167 @@
+
+        <TextView
+                android:id="@+id/tv_nick_name"
+                android:text="혼밥 시러~"
```
tools:text로 변경하고
android:text에는 databinding을 이용해 넣는 것이 어떨까요?

wjddnwls918 replied:
현재는 임시로 데이터를 넣은것이라 사용할때 수정이 필요할것 같습니다.

wjddnwls918 replied:
182a8713da1538d9b502f2a3eaa773a865493277

ro0opf commented on app/src/main/res/layout/board_preview_dialog.xml:
```diff
@@ -0,0 +1,167 @@
+        <TextView
+                android:id="@+id/tv_nick_name"
+                android:text="혼밥 시러~"
+                android:textSize="14sp"
```
textSize도 미리 dimens에 정의해놓는게 어떨까요?
@wjddnwls918 @Jiyunn

wjddnwls918 replied:
협의가 필요할것 같습니다. 이전에 얘기 나왔던 거긴하네요

wjddnwls918 committed:
#31-버전의 gradle project 수준 정의

wjddnwls918 committed:
#31-BoardPreviewDialog 수정

wjddnwls918 committed:
#31-버튼 리스너 삭제

wjddnwls918 committed:
Merge branch 'feature/#31-Board_Preview' of https://github.com/boostcampth/boostcamp3_D into feature/#31-Board_Preview

wjddnwls918 committed:
#31-android:text -> tools:text로 변경

ro0opf committed:
Merge branch 'develop' into feature/#31-Board_Preview

ro0opf committed:
Merge branch 'develop' into feature/#31-Board_Preview

wjddnwls918 committed:
#31-xml 셋팅 project -> default

wjddnwls918 committed:
#31-충돌 해결

# Conflicts:
#	app/src/main/res/layout/home_fragment.xml
#	app/src/main/res/layout/login_activity.xml
#	app/src/main/res/layout/main_activity.xml
#	app/src/main/res/layout/map_fragment.xml

wjddnwls918 committed:
#31-빌드 테스트

wjddnwls918 committed:
#31- 빌드 실패 문제 해결

wjddnwls918 committed:
#31-board_preview_dialog 레이아웃 수정

wjddnwls918 committed:
#31-circleImageView 크기 설정 문제

Zi-Yun commented on app/src/main/res/layout/board_preview_dialog.xml:
```diff
@@ -168,16 +168,18 @@
         android:src="@drawable/lion"
         app:civ_border_color="#FF000000"
         app:civ_border_width="2dp"
+        app:layout_constraintHeight_percent="0.3"
```
이 속성은 어떤 역할을 하는 건가요?

wjddnwls918 replied:
부모 뷰의 크기에 따라 비율을 고정하고 싶어 넣었습니다.

ro0opf committed:
Merge branch 'develop' into feature/#31-Board_Preview
